tests/unit/plugins/inventory/test_kubevirt_get_ssh_services_for_namespace.py

```diff
@@ -0,0 +1,168 @@
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {"name": "test-lb-ssh"},
+    "spec": {
+        "ports": [
+            {
+                "protocol": "TCP",
+                "port": 22,
+                "targetPort": 22,
+            },
+        ],
+        "type": "LoadBalancer",
+        "selector": {"kubevirt.io/domain": "test-lb-ssh"},
+    },
+}
+
+SVC_NP_SSH = {
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {"name": "test-np-ssh"},
+    "spec": {
+        "ports": [
+            {
+                "protocol": "TCP",
+                "port": 22,
+                "targetPort": 22,
+            },
+        ],
+        "type": "NodePort",
+        "selector": {"kubevirt.io/domain": "test-np-ssh"},
+    },
+}
+
+
+@pytest.mark.parametrize(
+    "client",
+    [
+        {
+            "services": [SVC_LB_SSH, SVC_NP_SSH],
+        },
+    ],
+    indirect=["client"],
+)
+def test_get_ssh_services_for_namespace(inventory, client):
+    assert inventory.get_ssh_services_for_namespace(client, DEFAULT_NAMESPACE) == {
+        "test-lb-ssh": SVC_LB_SSH,
+        "test-np-ssh": SVC_NP_SSH,
+    }
+
+
+SVC_CLUSTERIP = {
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {"name": "test-clusterip"},
+    "spec": {
+        "ports": [
+            {
+                "protocol": "TCP",
+                "port": 22,
+                "targetPort": 22,
+            },
+        ],
+        "type": "ClusterIP",
+        "selector": {"kubevirt.io/domain": "test-clusterip"},
+    },
+}
+
+
+SVC_HTTP = {
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {"name": "test-http"},
+    "spec": {
+        "ports": [
+            {
+                "protocol": "TCP",
+                "port": 80,
+                "targetPort": 80,
+            },
+        ],
+        "type": "LoadBalancer",
+        "selector": {"kubevirt.io/domain": "test-http"},
+    },
+}
+
+
+SVC_NO_SPEC = {
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {"name": "test-no-spec"},
+}
+
+SVC_NO_TYPE = {
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {"name": "test-no-type"},
+    "spec": {
+        "ports": [
+            {
+                "protocol": "TCP",
+                "port": 22,
+                "targetPort": 22,
+            },
+        ],
+        "selector": {"kubevirt.io/domain": "test-no-type"},
+    },
+}
+
+SVC_NO_PORTS = {
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {"name": "test-no-ports"},
+    "spec": {
+        "type": "LoadBalancer",
+        "selector": {"kubevirt.io/domain": "test-no-ports"},
+    },
+}
+
+SVC_NO_SELECTOR = {
+    "apiVersion": "v1",
+    "kind": "Service",
+    "metadata": {"name": "test-no-selector"},
+    "spec": {
+        "ports": [
+            {
+                "protocol": "TCP",
+                "port": 22,
+                "targetPort": 22,
+            },
+        ],
+        "type": "LoadBalancer",
+    },
+}
```

Comment:
Could we reduce these definitions using a base service and using `merge_dicts`?

Reply:
I feel like it makes things more complicated as they need to be. Also, I think it was not possible to merge the `spec` of two dict this way.